### PR TITLE
SDC Behavior Overhaul

### PR DESCRIPTION
**Pull request by kareefardi**
```
+ Add PNR_SDC_FILE
+ Warn when (PNR|SIGNOFF)_SDC_FILE are not overwritten by the user.
+ Add SDC_IN and DEFAULT_SDC_FILE to ignore list in CI
+ Add per corner max slew/fanout/cap count to sta report
+ Add per corner worst hold and setup value to sta report 
~ Rename RCX_SDC_FILE to SIGNOFF_SDC_FILE.
~ Always use PNR_SDC_FILE instead of CURRENT_SDC except during signoff stage, where SIGNOFF_SDC_FILE is used instead
~ Enable DIODE_ON_PORTS and RUN_HEURISTIC_DIODE_INSERTION for APU which fails after SDC updates.
~ Adjust timing checkers according to the new reported values 
```

### CODE REVIEW

File: configuration/general.tcl [hunk 1]
```diff
@@ -80,3 +80,9 @@ set ::env(RUN_KLAYOUT_XOR) 1
 set ::env(GENERATE_FINAL_SUMMARY_REPORT) {1}
 
 set ::env(WRITE_VIEWS_NO_GLOBAL_CONNECT) 0
+
+
+set ::env(BASE_SDC_FILE) $::env(SCRIPTS_DIR)/base.sdc
+set ::env(DEFAULT_SDC_FILE) $::env(BASE_SDC_FILE)
```
Comment (donn) [inline configuration/general.tcl:86]: What does `DEFAULT_SDC_FILE` do?

File: docs/source/reference/configuration.md [hunk 2]
```diff
@@ -303,7 +304,7 @@ These variables worked initially, but they were too sky130 specific and will be
 | `RUN_SPEF_EXTRACTION` <a id="RUN_SPEF_EXTRACTION"></a> | Specifies whether or not to run SPEF extraction on the routed DEF. 1=enabled 0=disabled <br> (Default: `1`) |
 | `SPEF_EXTRACTOR` <a id="SPEF_EXTRACTOR"></a> | Specifies which spef extractor to use. Values: `openrcx` or (**removed:** `def2spef`). <br> (Default: `openrcx`) |
 | `RCX_MERGE_VIA_WIRE_RES` <a id="RCX_MERGE_VIA_WIRE_RES"></a> | Specifies whether to merge the via resistance with the wire resistance or separate it from the wire resistance. 1 = Merge via resistance, 0 = Separate via resistance <br> (Default: `1`)|
-| `RCX_SDC_FILE` <a id="RCX_SDC_FILE"></a> | Specifies SDC file to be used for RCX-based STA, which can be different from the one used for implementation. <br> (Default: `BASE_SDC_FILE`) |
+| `RCX_SDC_FILE` <a id="RCX_SDC_FILE"></a> | **Deprecated: Use SIGNOFF_SDC_FILE.**: Specifies SDC file to be used for RCX-based STA, which can be different from the one used for implementation. <br> (Default: `BASE_SDC_FILE`) |
```
Comment (donn) [inline docs/source/reference/configuration.md:307]: wrap `SIGNOFF_SDC_FILE` in \`\` and no ending comma

File: scripts/openroad/common/io.tcl [hunk 3]
```diff
@@ -90,9 +90,7 @@ proc read_netlist {args} {
 
     link_design $::env(DESIGN_NAME)
 
-    if { [info exists ::env(CURRENT_SDC)] } {
-        read_current_sdc
-    }
+    read_current_sdc
```
Comment (donn) [inline scripts/openroad/common/io.tcl:93]: good

File: scripts/tcl_commands/synthesis.tcl [hunk 4]
```diff
@@ -113,6 +113,7 @@ proc run_synthesis {args} {
     puts_info "Running Synthesis (log: [relpath . $log])..."
 
     set ::env(CURRENT_SDC) $::env(BASE_SDC_FILE)
+    set ::env(SDC_IN) $::env(BASE_SDC_FILE)
```
Comment (donn) [inline scripts/tcl_commands/synthesis.tcl:116]: It appears that you're relying on the flow being run as a monolith for `SDC_IN` to have an appropriate value, i.e., if synthesis is not run then multiple PnR steps will fail. I'm wondering if it's appropriate to move the logic of picking between signoff/PnR SDC inside `read_current_sdc`

**Review by donn — APPROVED** [review]
Looks fine now

File: scripts/tcl_commands/sta.tcl [hunk 5]
```diff
@@ -157,22 +151,21 @@ proc run_parasitics_sta {args} {
...
-                    -tool sta
-
-                set ::env(LAST_TIMING_REPORT_TAG) [index_file $::env(signoff_reports)/rcx_sta]
+                lappend sta_flags -blackbox_check
+            }
+
+            run_sta {*}$sta_flags
+
+            if { $process_corner == "nom" } {
+                set ::env(LAST_TIMING_REPORT_TAG) "[index_file $::env(signoff_reports)/sta-rcx_$process_corner]/multi_corner_sta"
```
Comment (donn) [inline scripts/tcl_commands/sta.tcl:168]: WAIT. Are we sure this extracts the right value? There are three values in each report.

**Review by donn — CHANGES_REQUESTED** [review]

File: scripts/tcl_commands/sta.tcl (hunk 5 above)
Comment (kareefardi) [reply scripts/tcl_commands/sta.tcl:168]: There were always three values in each report. 
The previous code runs the same way:
```
                run_sta\
                    -log $::env(signoff_logs)/rcx_sta.log\
                    -process_corner $process_corner\
                    -save_to $directory \
                    -blackbox_check \
                    -tool sta
```

Comment (donn) [reply scripts/tcl_commands/sta.tcl:168]: `run_sta` is single corner

**Review by donn — APPROVED** [review]

**Pull request merged**